src/main/java/com/exactpro/th2/sailfish/utils/ProtoToIMessageConverter.java

```diff
@@ -151,6 +151,9 @@ private Object traverseFilterField(String fieldname, ValueFilter value) {
         if (value.hasMessageFilter()) {
             return fromProtoFilter(value.getMessageFilter(), fieldname);
         }
+        if (value.getOperation() == FilterOperation.IN || value.getOperation() == FilterOperation.NOT_IN) {
+            return new ListContainFilter(value.getOperation(), value.getSimpleList().getSimpleValuesList());
+        }
```

Comment:
You should check if it has a simple list first and then if it also has the right operation - create the filter. Otherwise, you should report an error